subsys/shell/shell.c

```diff
@@ -1072,22 +1097,20 @@ static void shell_log_process(const struct shell *shell)
 
 		processed = shell_log_backend_process(shell->log_backend);
 
-		if (shell->ctx->state != SHELL_STATE_COMMAND) {
-			struct k_poll_signal *signal =
-				&shell->ctx->signals[SHELL_SIGNAL_RXRDY];
-
-			cmd_line_print(shell);
+		struct k_poll_signal *signal =
+			&shell->ctx->signals[SHELL_SIGNAL_RXRDY];
 
-			/* Arbitrary delay added to ensure that prompt is
-			 * readable and can be used to enter further commands.
-			 */
-			if (shell->ctx->cmd_buff_len) {
-				k_sleep(K_MSEC(15));
-			}
+		cmd_line_print(shell);
 
-			k_poll_signal_check(signal, &signaled, &result);
+		/* Arbitrary delay added to ensure that prompt is
+		 * readable and can be used to enter further commands.
+		 */
+		if (shell->ctx->cmd_buff_len) {
+			k_sleep(K_MSEC(15));
```

Comment:
Should this be made configurable? Or even better, could you use perhaps `k_yield()` instead?

Reply:
It is not related to this change. @nordic-krch  what is your opinion about it?

Reply:
No need to move to Kconfig now if we haven't seen any issues yet with it not being configurable. Thanks for the explanation.